libmscore/spanner.cpp

```diff
@@ -304,6 +304,12 @@ Spanner::Spanner(const Spanner& s)
 
 Spanner::~Spanner()
       {
+      for (SpannerSegment* ss : segments)
+            if (ss->system())
+                  ss->system()->remove(ss);
+      for (SpannerSegment* ss : unusedSegments)
+            if (ss->system())
+                  ss->system()->remove(ss);
```

Comment:
This smells like another crash I was seeing…

Reply:
Me too, crash in SpannerSegment destructor on closing a score

Reply:
Concerning these changes (this and the one in Score destructor) it is also possible to fix this by making systems be deleted before other elements: https://github.com/musescore/MuseScore/pull/4278/commits/e85f3a218cd275f003504044ce73bca3416be23c. Not sure which approach is better though.

Reply:
In any case, we have to make sure that deleting the system does not try to delete objects belonging to the score.
I also don't know which approach would be the better.
The important thing is to fix bug, somehow :-)